awslabs/aws-lambda-powertools-typescript
# refactor(parameters): moved ssm resource creation to `AwsCustomResource`

## Description of your changes

As described in the linked issue (#1316), as part of the integration tests for the `SSMProvider` part of the Parameters utility we are creating some SecureString SSM resources. The creation of this type of resources was initially implemented with a Lambda-backed custom resource because CDK doesn't offer any L2 construct for them.

This PR refactor that portion of code to instead use `AwsCustomResource`. This allows to remove a significant amount of code and to not have to maintain a dedicated Lambda function whose single purpose was to create & delete these resources. With the usage of `AwsCustomResource` the responsibility of this function is moved to CDK instead.

Once merged, this PR will close #1316.

### How to verify this change

See successful e2e test run [here](https://github.com/awslabs/aws-lambda-powertools-typescript/actions/runs/4234303153).

### Related issues, RFCs



**Issue number:** #1316

### PR status

***Is this ready for review?:*** YES  
***Is it a breaking change?:*** NO

## Checklist


### Breaking change checklist


---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
